### DISCUSSION TITLE
beta.foldingathome.org - settings & capabilities with 8.x

### DISCUSSION THREAD

**Discussion by CHDAFNI-MSFT** [FoldingAtHome/fah-client-bastet#249] [Q&A]
Currently on latest alpha 8.3.17

headless builds - in my scenario the workers are ephemeral - they will spin up brand new each time and are deleted automatically when there's resource exhaustion in the hosting environment.

---
When adding a worker, I noticed the following:
1. The GPU is not enabled by default - even when detected/supported - I must manually 'check' the box/save
2. Folding / Start - must be manually initiated from the C&C (command and control) portal - (beta.foldingathome.org) - else it sits idle

In 7.x these were set as part of the config.xml

Questions:
1. Is it possible in 8.x with config.xml - to auto start and include GPU?
2. Rather than configuring the settings via config.xml - will the C&C portal in the future have 'global settings' or similar capability for applying settings to group(s) of machines automatically?

**Comment by muziqaz** [comment]
I believe issue 1 has been designed by request from us. For starters GPUs, especially in Linux are problematic, and even in Windows if user unknown to them, installs fahclient with MS pre-installed GPU driver, which has no real functionality except to display desktop, automatic GPU enablement and start will waste a lot of WUs before user catches on it.
Autostart in general is terrible idea. By the time user gets to grips what is happening, fahclient will download at minimum 1 WU, and if you have a GPU and iGPU on the system, that is 3 WUs downloaded without users acknowledgement.
Another scenario: user install v8, and wants to not use GPU, since it is unstable for them. Why would fahclient assume that user wants to contribute with everything they got. Leave that to a user to decide
One more scenario: old fart like me, who wants every foding device inside of the system to be configured into custom resource groups (since I grew up with Slots on v7). If we enable autoenablment and autostart, I loce that option to configure things up to my liking, because fahclient already downloaded work and is folding.
Besides, after fahclient installation web control launches, which should have this Play button in the centre. I'm sure user these days know that this would mean: Start folding. 
This topic has been debated internally quite a bit. Believe it or not Joe had it set up as auto start and auto enable in the beginning. But was outvoted by us :D

**Comment by kbernhagen** [comment]
It may not matter for these issues, but for alpha clients, you should use
https://alpha.foldingathome.org/

**Reply by CHDAFNI-MSFT** [reply]
my bad and thank you! I do see my login/icon in this portal

**Comment by kbernhagen** [comment]
Your issues would be solved by using persistent fah client data directories that are uniquely used when spinning up a container.

It is important that folding finishes and pauses before a container exits, because the v8 client will dump current work if it detects a machine id change.

**Comment by kbernhagen** [comment]
See also https://github.com/FoldingAtHome/fah-client-bastet/issues/216

**Comment by kbernhagen** [comment]
To send commands by script after a container v8 client starts, this may be helpful:
https://github.com/kbernhagen/lufah
If you put your GPU in a separate resource group, you can send fold/pause/finish to that group.

See also https://github.com/FoldingAtHome/fah-client-bastet/issues/119

**Comment by CHDAFNI-MSFT** [comment]
I'll clarify my hopes and intentions.

I do not want to change any default out of the box behavior - I completely understand where you're coming from with the perspective of the new or average F@H user. I am the exception, not the norm.

I'm seeking what is, may be, or will be possible.

I don't think the original Q's were addressed but just know that for areas that aren't the default - can it instead be configured via the config.xml or the C&C portal via a future 'global setting'.

Today with 7.x - (with drivers, packages, and a pre-made config.xml automatically loaded on system first boot) - my F@H clients roar away without any manual intervention - it's what has allowed me to be in the top 100.

**Reply by kbernhagen** [reply]
Answer to your question 1 is no and no.

Currently, all global settings are in the Web Control Account Settings page.
If there are additional/optional settings you would like to be global, please make an enhancement request.

As to what you want moving from v7 to v8, you need to be very specific.
For instance,
What container or other system are you using?
How do you dynamically adjust allocated resources?
How do you persist the data directories/config.xml now?

**Comment by kbernhagen** [comment]
Normally, your config.xml will only need account-token to use all the global account settings and be controllable by remote Web Control.

See https://github.com/FoldingAtHome/fah-client-bastet/discussions/214

**Comment by kbernhagen** [comment]
Do you use the v7 exit-when-done option?
